handlers/utils_test.go

```diff
@@ -68,3 +69,242 @@ func TestUtils(t *testing.T) {
 	assert.Equal(t, len(jobConfigs), 2)
 
 }
+
+// Test that CheckIndexingJobsImageConfig does not return an error when both
+// Indexd and Metadata Service creds have been configured.
+func TestCheckIndexingJobsImageConfigWithIndexdAndMDSCreds(t *testing.T) {
+	jobsJson :=
+		`
+	[
+	  {
+		"name": "indexing",
+		"pattern": "s3://xssxs/*",
+		"image": "quay.io/cdis/indexs3client:master",
+		"imageConfig": {
+		  "url": "http://indexd-service/",
+		  "username": "test",
+		  "password": "test",
+		  "metadataService": {
+		    "url": "http://revproxy-service/mds",
+		    "username": "dog",
+		    "password": "paws"
+		  }
+		}
+	  },
+	  {
+		"name": "usersync",
+		"pattern": "s3://xssxs/user.yaml",
+		"image": "quay.io/cdis/fence:master",
+		"imageConfig": {}
+	  }
+	]
+	`
+	jobConfigs := make([]JobConfig, 0)
+	json.Unmarshal([]byte(jobsJson), &jobConfigs)
+	err := CheckIndexingJobsImageConfig(jobConfigs)
+	assert.Equal(t, err, nil)
+}
+
+// Test that CheckIndexingJobsImageConfig returns an error when MDS creds
+// have not been configured.
```

Comment:
i don't get why we're making this non backwards compatible - what if we need ssjdispatcher in a commons that doesn't use MDS?

Reply:
This is meant as an extra precaution against the latest SSJDispatcher/indexs3client being deployed to a commons without MDS creds configured for SSJDispatcher, causing this new feature not to work. It's intended to be an automated way to signal to DevOps that SSJDispatcher/indexs3client are not healthy.

If SSJDispatcher is needed in a commons that doesn't use MDS, would it be reasonable to just use an older version of SSJDispatcher?

Also, one thing to keep in mind with this PR is that it only requires MDS creds for `"name": "indexing"` jobs, so it's not going to affect SSJDispatcher deployments in which there is no `indexing` job configured (not sure if that's the case for any commons).

Reply:
> If SSJDispatcher is needed in a commons that doesn't use MDS, would it be reasonable to just use an older version of SSJDispatcher?

no, i don't think so. The ability to upload or index data (SSJDispatcher) is not dependent on whether the MDS is used. MDS is an optional service; all Commons should be able to get the latest SSJDispatcher patches regardless of whether they use it

Reply:
we need to have the backward compatibility. Some commons do not require MDS

Reply:
mds is no longer optional
there's no reason to not run mds
it adds complexity and new ways to eff up if we make it optional

Reply:
then @johnfrancismccann can you update your "Deployment changes" to make it very clear that we need to add the MDS to the manifest

Reply:
For sure, I'll make note of that in "Deployment changes" in the [indexs3client PR](https://github.com/uc-cdis/indexs3client/pull/29) as well.